### PR TITLE
Refactor country_dashboard.py

### PR DESCRIPTION
**Pull request by lukipuki**
Create DashboardFactory, which:

-  Creates important data structures in its constructor.
- Once initialized, can create all the different dashboards.

### CODE REVIEW

File: covid_graphs/covid_graphs/scripts/country_dashboard.py [hunk 1]
```diff
@@ -46,17 +46,14 @@ def __init__(self, dashboard_type: DashboardType, data_dir: Path, server: Flask)
...
-            dashboard_type == DashboardType.SingleCountry
-            or dashboard_type == DashboardType.AllCountries
-        ):
-            self.graph_dict = {
-                prediction_event.name: CountryDashboard._create_graphs(
-                    self.report_by_short_name, prediction_event
-                )
-                for prediction_event in prediction_db.get_prediction_events()
-            }
+        self.graph_dict = {
```
Comment (miskosz) [inline covid_graphs/covid_graphs/scripts/country_dashboard.py]: What do you think about `self.graphs_by_event`?

File: covid_graphs/covid_graphs/scripts/country_dashboard.py [hunk 2]
```diff
@@ -94,7 +112,7 @@ def _create_graphs(
     @staticmethod
     def _create_buttons(
         dashboard_type: DashboardType, report_by_short_name: Dict[str, CountryReport]
-    ):
+    ) -> List[dash.development.base_component.Component]:
```
Comment (miskosz) [inline covid_graphs/covid_graphs/scripts/country_dashboard.py:115]: 💯

**Review by miskosz — APPROVED** [review]
LGTM

File: covid_graphs/covid_graphs/scripts/country_dashboard.py (hunk 1 above)
Comment (lukipuki) [reply covid_graphs/covid_graphs/scripts/country_dashboard.py]: Done

**Pull request merged**